### PR TITLE
Bug 1225057 - Rewrite NotificationCollection/NotificationModel to be a store

### PR DESCRIPTION
**Pull request by malayaleecoder**
Part 1 of the bug. 
@Standard8 : Please have a look.

### CODE REVIEW

**Comment by Standard8** [comment]
It looks like you missed a change in `panel.html` for updating the directory of `models.js` - when I did a `make test` locally the functional tests failed (at the moment the tests aren't run on Travis that reports to the PR).

Can you also change the commit message to be something like `Bug 1225057 - Part 1. Move the notification models and views into the desktop specific directory. r=Standard8`

I think once that is done, we should be there!

**Comment by malayaleecoder** [comment]
@Standard8 : I noticed the Error on `make test` but thought that it would be due to the fact that it was not able to open Chrome. Anyway, I have made the changes, but still observe the same error. Please have a look.

File: add-on/panels/panel.html [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
     <script type="text/javascript" src="shared/js/loopapi-client.js"></script>
     <script type="text/javascript" src="shared/js/utils.js"></script>
-    <script type="text/javascript" src="shared/js/models.js"></script>
+    <script type="text/javascript" src="add-on/panels/js/models.js"></script>
```
Comment (Standard8) [inline add-on/panels/panel.html]: This needs to be just `panels/js/models.js` (see the paths for some other panels files further down in the file).

**Comment by malayaleecoder** [comment]
@Standard8 Sorry, did not notice that in a hurry.

**Comment by Standard8** [comment]
That's great. r=Standard8

**Pull request merged**